chain/rust/src/builders/tx_builder.rs

```diff
@@ -0,0 +1,4966 @@
+    }
+}
+
+// tx_body must be the result of building from tx_builder
+// constructs the rest of the Transaction using fake witness data of the correct length
+// for use in calculating the size of the final Transaction
+fn fake_full_tx(tx_builder: &TransactionBuilder, body: TransactionBody) -> Result<Transaction, JsError> {
+    Ok(Transaction {
+        body,
+        witness_set: tx_builder.witness_builders.build_fake(),
+        is_valid: true,
+        auxiliary_data: tx_builder.auxiliary_data.clone(),
+    })
+}
+
+fn min_fee(tx_builder: &TransactionBuilder) -> Result<Coin, JsError> {
+    let full_tx = fake_full_tx(tx_builder, tx_builder.build_body()?)?;
+    // we can't know the of scripts yet as they can't be calculated until we build the tx
+    ledger::alonzo::fees::min_no_script_fee(&full_tx, &tx_builder.config.fee_algo)
+}
+
+fn min_fee_with_exunits(tx_builder: &TransactionBuilder) -> Result<Coin, JsError> {
+    let full_tx = fake_full_tx(tx_builder, tx_builder.build_body()?)?;
+    // we can't know the of scripts yet as they can't be calculated until we build the tx
+    ledger::alonzo::fees::min_fee(&full_tx, &tx_builder.config.fee_algo, &tx_builder.config.ex_unit_prices)
+}
+
+#[wasm_bindgen]
+pub enum CoinSelectionStrategyCIP2 {
+    /// Performs CIP2's Largest First ada-only selection. Will error if outputs contain non-ADA assets.
+    LargestFirst,
+    /// Performs CIP2's Random Improve ada-only selection. Will error if outputs contain non-ADA assets.
+    RandomImprove,
+    /// Same as LargestFirst, but before adding ADA, will insert by largest-first for each asset type.
+    LargestFirstMultiAsset,
+    /// Same as RandomImprove, but before adding ADA, will insert by random-improve for each asset type.
+    RandomImproveMultiAsset,
+}
+
+#[wasm_bindgen]
+#[derive(Clone, Debug)]
+pub struct TransactionBuilderConfig {
+    fee_algo: LinearFee,
+    pool_deposit: BigNum,      // protocol parameter
+    key_deposit: BigNum,       // protocol parameter
+    max_value_size: u32,       // protocol parameter
+    max_tx_size: u32,          // protocol parameter
+    /// TODO: remove once Babbage is on mainnet
+    coins_per_utxo_word: Option<Coin>, // protocol parameter
+    coins_per_utxo_byte: Coin, // protocol parameter
+    ex_unit_prices: ExUnitPrices, // protocol parameter
+    costmdls: Costmdls,           // protocol parameter
+    collateral_percentage: u32,   // protocol parameter
+    max_collateral_inputs: u32,   // protocol parameter
+    prefer_pure_change: bool,
+}
+
+#[wasm_bindgen]
+#[derive(Clone, Debug, Default)]
+pub struct TransactionBuilderConfigBuilder {
+    fee_algo: Option<LinearFee>,
+    pool_deposit: Option<BigNum>,      // protocol parameter
+    key_deposit: Option<BigNum>,       // protocol parameter
+    max_value_size: Option<u32>,       // protocol parameter
+    max_tx_size: Option<u32>,          // protocol parameter
+    /// TODO: remove once Babbage is on mainnet
```

Comment:
babbage is on mainnet already

Reply:
yeah, this come from the old code that was copy-pasted. No need to change it in this PR

Reply:
issue for that -- https://github.com/dcSpark/cardano-multiplatform-lib/issues/200